solutions/src/test/java/ru/sberbank/school/task11/MyStreamsTest.java

```diff
@@ -0,0 +1,72 @@
+package ru.sberbank.school.task11;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+class MyStreamsTest {
+
+    @Test
+    @DisplayName("Тест на выброс NP при передаче null коллекции")
+    void of() {
+        List<String> list = null;
+        Assertions.assertThrows(NullPointerException.class, () -> MyStreams.of(list));
+    }
+
+    @Test
+    void toSet() {
+        Set<String> strings = MyStreams.of("apple", "banana", "orange", "lemon", "strawberry", "lemon", "strawberry").toSet();
+        Set<String> set = Stream.of("apple", "banana", "orange", "lemon", "strawberry", "lemon", "strawberry").collect(Collectors.toSet());
+
+        Assertions.assertIterableEquals(set, strings);
+    }
+
+    @Test
+    void toList() {
+        List<String> list = MyStreams.of("apple", "banana", "orange", "lemon", "strawberry").toList();
+
+        Assertions.assertIterableEquals(Arrays.asList("apple", "banana", "orange", "lemon", "strawberry"), list);
+    }
+
+    @Test
+    void filter() {
+        List<String> list = MyStreams.of("apple", "banana", "orange", "lemon", "strawberry").filter(s -> s.length() == 5).toList();
+
+        Assertions.assertIterableEquals(Arrays.asList("apple", "lemon"), list);
+    }
+
+    @Test
+    void transform() {
+        List<Integer> integerList = MyStreams.of("apple", "banana", "orange", "lemon", "strawberry", "lemon", "strawberry").transform(String::length).toList();
+        List<Integer> integerList1 = Stream.of("apple", "banana", "orange", "lemon", "strawberry", "lemon", "strawberry").map(String::length).collect(Collectors.toList());
+
+        Assertions.assertIterableEquals(integerList, integerList1);
+    }
+
+    @Test
+    void sorted() {
+        List<String> list = MyStreams.of("lemon", "strawberry", "apple", "banana", "orange", "lemon", "strawberry").sorted(String::compareTo).toList();
+        List<String> list1 = Stream.of("lemon", "strawberry", "apple", "banana", "orange", "lemon", "strawberry").sorted(String::compareTo).collect(Collectors.toList());
```

Comment:
Line is longer than 120 characters (found 169).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-821069581>*